pkg/ipamd/ipamd.go

```diff
@@ -1920,3 +1924,30 @@ func (c *IPAMContext) getPrefixesNeeded() int {
 	log.Debugf("ToAllocate: %d", toAllocate)
 	return toAllocate
 }
+
+func (c *IPAMContext) CheckEC2Permissions(nodeInitErr error) error {
+	if strings.Contains(nodeInitErr.Error(), "UnauthorizedOperation") {
+		ctx := context.TODO()
+
+		listOptions := client.ListOptions{}
+		fieldSelector := client.MatchingFields{"spec.nodeName": c.myNodeName}
+		labelSelector := client.MatchingLabels{"k8s-app": "aws-node"}
+		fieldSelector.ApplyToList(&listOptions)
+		labelSelector.ApplyToList(&listOptions)
+
+		var podList corev1.PodList
+		err := c.rawK8SClient.List(ctx, &podList, &listOptions)
+		if err != nil {
+			log.Errorf("Failed to get pods: %v", err)
+			return err
+		}
+
+		for _, pod := range podList.Items {
+			log.Debugf("Broadcast event on pod %s", pod.Name)
+			printErr := strings.Replace(nodeInitErr.Error(), "\n", "", 1)
```

Comment:
Could you also print the event in the PR description. Does this print the encoded message in the event too?

IMO, something like this would be more descriptive and not mention the "node" role as the user could also be using IRSA.

*Reason:* MissingIAMPolicy
*Message:* "aws-node failed to start because it's unauthorized to perform EC2 API calls. Please verify AmazonEKS_CNI_Policy is attached to the supplied IAM Role" 

I think if we send out the entire error message in Pod event, it will not get de-duped and be shown as separate events. However, the warning above will get de-dupeed like this - 37m (x19 over 18h)